src/statement.js

```diff
@@ -193,8 +210,9 @@ pp.parseForStatement = function(node) {
 }
 
 pp.parseFunctionStatement = function(node) {
+  let isAsync = this.options.ecmaVersion >= 8 && this.eatContextual("async")
```

Comment:
I'd prefer for the callee to be responsible for passing `isAsync` and skipping the `async` token (since it always already knows whether this is an async function, there's no need to repeat the check).